connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -53,36 +56,68 @@
 
     private Map<String, String> config;
     private State state;
+    private final OffsetStorageReader offsetStorageReader;
 
-    public WorkerConnector(String connName,
-                           Connector connector,
-                           ConnectorContext ctx,
-                           ConnectorStatus.Listener statusListener) {
+    public WorkerConnector(final String connName,
+                           final Connector connector,
+                           final ConnectorContext ctx,
+                           final ConnectorStatus.Listener statusListener,
+                           final OffsetStorageReader offsetStorageReader) {
         this.connName = connName;
         this.ctx = ctx;
         this.connector = connector;
         this.statusListener = statusListener;
         this.state = State.INIT;
+        this.offsetStorageReader = offsetStorageReader;
     }
 
     public void initialize(ConnectorConfig connectorConfig) {
         try {
             this.config = connectorConfig.originalsStrings();
             log.debug("{} Initializing connector {} with config {}", this, connName, config);
-
-            connector.initialize(new ConnectorContext() {
+            final ConnectorContext delegateCtx = new ConnectorContext() {
                 @Override
                 public void requestTaskReconfiguration() {
-                    ctx.requestTaskReconfiguration();
+                    WorkerConnector.this.ctx.requestTaskReconfiguration();
                 }
 
                 @Override
                 public void raiseError(Exception e) {
                     log.error("{} Connector raised an error", this, e);
```

Comment:
I know this line hasn't been changed, but the `this` variable refers to the `ConnectorContext` that does not have a `toString()` method. Might be better to refer to `WorkerConnector.this` instead, so that it behaves like old line 87.